tools/pipeline_perf_test/orchestrator/orchestrator.py

```diff
@@ -63,6 +204,89 @@ def shutdown(self) -> None:
             print(f"Error deleting Kubernetes resources: {e}")
             print(f"Error output: {e.stderr}")
 
+    def start_monitoring(self, interval: float) -> None:
+        """
+        Monitor a Kubernetes pod's CPU and memory usage in a background thread using subprocess.
+
+        Args:
+            interval: Time in seconds between polling.
+        """
+        def monitor(deployment_name: str,
+                    namespace: str,
+                    stats: ProcessStats,
+                    stop_event: threading.Event,
+                    interval: float = 1.0):
+            # The approach below relies on kubernetes monitoring-server and kubelet metrics via
+            # cAdvisor. These have an update interval on the order of 15 seconds, which is
+            # fairly high for short lived tests. We will likely need to evaluate other approaches.
+            # Possibly lowering housekeeping interval:
+            # https://github.com/google/cadvisor/issues/2660 but that's also not a great solution.
+            def get_pod_name():
+                # Get the pod name associated with the deployment
+                try:
+                    cmd = [
+                        "kubectl", "get", "pods", "-n", namespace,
+                        "-l", f"app={deployment_name}",
+                        "-o", "jsonpath={.items[0].metadata.name}"
+                    ]
+                    pod_name = subprocess.check_output(cmd, text=True).strip()
+                    return pod_name
+                except subprocess.CalledProcessError as e:
+                    print(f"Failed to get pod for deployment {deployment_name}: {e}")
+                    return None
+
+            pod_name = get_pod_name()
+            if not pod_name:
+                print("Pod name could not be determined. Monitoring aborted.")
+                return
+
+            while not stop_event.is_set():
+                try:
+                    cmd = ["kubectl", "top", "pod", pod_name, "-n", namespace, "--no-headers"]
+                    result = subprocess.check_output(cmd, text=True).strip()
+
+                    if result:
+                        # Example: mypod-abc123 12m 21Mi
+                        parts = result.split()
+                        cpu_str = parts[1]  # e.g., '12m'
+                        mem_str = parts[2]  # e.g., '21Mi'
+
+                        # Convert CPU and memory to standardized formats
+                        if cpu_str.endswith('m'):
```

Comment:
if the Cpu stats from K8s same format as docker? Docker seems to be percentage utilization, and this one seems different?

Reply:
See above for more detail, this is cores or milicores consumed 0-1 * num_cores